Build failure: quodlibet-full

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

I updated my system to the latest version:
```bash
sudo nix flake update --flake ~/.config/nixos-config/
sudo nixos-rebuild switch --flake ~/.config/nixos-config/
``` 

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/300737477

### Relevant log output

```console
bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8): No such file or directory
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pypa-build-hook
Using pypaBuildPhase
Sourcing python-runtime-deps-check-hook
Using pythonRuntimeDepsCheckHook
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing sphinx-hook
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/bdycn0jjsmvmf5h7zl2cr0c0kizs9k66-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file "source/tests/test_windows.py"
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/sy6szni1j7jbb26nr4bdhw24s0c4lnf6-fix-gdist-python-3.12.patch
patching file gdist/__init__.py
patch unexpectedly ends in middle of line
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing pypaBuildPhase
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...
warning: no files found matching 'HACKING'
warning: no files found matching 'data/*.sh'
no previously-included directories found matching 'docs/_build_guide'
no previously-included directories found matching 'docs/_build_all'
no previously-included directories found matching 'docs/_rtd_theme*'
* Building wheel...
/build/source/gdist/gettextutil.py:179: GettextWarning: xgettext: warning: a fallback ITS rule file '/nix/store/zs5crhr67zp8cxn7dh4mwq08zw3sb31m-gettext-0.22.5/share/>
xgettext: warning: a fallback ITS rule file '/nix/store/zs5crhr67zp8cxn7dh4mwq08zw3sb31m-gettext-0.22.5/share/gettext-0.22.5/its/metainfo.its' is used; it may not be >

  pot_path = _create_pot(potfiles_path, src_root)
warning: no files found matching 'HACKING'
warning: no files found matching 'data/*.sh'
no previously-included directories found matching 'docs/_build_guide'
no previously-included directories found matching 'docs/_build_all'
no previously-included directories found matching 'docs/_rtd_theme*'
gtk-update-icon-cache: No theme index file.
Successfully built quodlibet-4.7.dev0-py3-none-any.whl
Finished creating a wheel...
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
@nix { "action": "setPhase", "phase": "pythonRuntimeDepsCheckHook" }
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for quodlibet-4.7.dev0-py3-none-any.whl
Finished executing pythonRuntimeDepsCheck
Running phase: glibPreInstallPhase
@nix { "action": "setPhase", "phase": "glibPreInstallPhase" }
Running phase: glibPreInstallPhase
@nix { "action": "setPhase", "phase": "glibPreInstallPhase" }
Running phase: glibPreInstallPhase
@nix { "action": "setPhase", "phase": "glibPreInstallPhase" }
Running phase: glibPreInstallPhase
@nix { "action": "setPhase", "phase": "glibPreInstallPhase" }
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Executing pypaInstallPhase
Successfully installed quodlibet-4.7.dev0-py3-none-any.whl
Finished executing pypaInstallPhase
Running phase: pythonOutputDistPhase
@nix { "action": "setPhase", "phase": "pythonOutputDistPhase" }
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
Running phase: gappsWrapperArgsHook
@nix { "action": "setPhase", "phase": "gappsWrapperArgsHook" }
Running phase: dropIconThemeCache
@nix { "action": "setPhase", "phase": "dropIconThemeCache" }
Running phase: dropIconThemeCache
@nix { "action": "setPhase", "phase": "dropIconThemeCache" }
Running phase: dropIconThemeCache
@nix { "action": "setPhase", "phase": "dropIconThemeCache" }
Running phase: fixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
Symlinking parent icon themes...
Symlinking parent icon themes...
Wrapping program '/nix/store/z187sx56kr4h6ypl33mx4p7qjk2ppznr-quodlibet-full-4.6.0-unstable-2024-08-08/bin/exfalso'
Wrapping program '/nix/store/z187sx56kr4h6ypl33mx4p7qjk2ppznr-quodlibet-full-4.6.0-unstable-2024-08-08/bin/operon'
Wrapping program '/nix/store/z187sx56kr4h6ypl33mx4p7qjk2ppznr-quodlibet-full-4.6.0-unstable-2024-08-08/bin/quodlibet'
shrinking RPATHs of ELF executables and libraries in /nix/store/z187sx56kr4h6ypl33mx4p7qjk2ppznr-quodlibet-full-4.6.0-unstable-2024-08-08
shrinking /nix/store/z187sx56kr4h6ypl33mx4p7qjk2ppznr-quodlibet-full-4.6.0-unstable-2024-08-08/bin/exfalso
shrinking /nix/store/z187sx56kr4h6ypl33mx4p7qjk2ppznr-quodlibet-full-4.6.0-unstable-2024-08-08/bin/operon
shrinking /nix/store/z187sx56kr4h6ypl33mx4p7qjk2ppznr-quodlibet-full-4.6.0-unstable-2024-08-08/bin/quodlibet
checking for references to /build/ in /nix/store/z187sx56kr4h6ypl33mx4p7qjk2ppznr-quodlibet-full-4.6.0-unstable-2024-08-08...
gzipping man pages under /nix/store/z187sx56kr4h6ypl33mx4p7qjk2ppznr-quodlibet-full-4.6.0-unstable-2024-08-08/share/man/
patching script interpreter paths in /nix/store/z187sx56kr4h6ypl33mx4p7qjk2ppznr-quodlibet-full-4.6.0-unstable-2024-08-08
stripping (with command strip and flags -S -p) in  /nix/store/z187sx56kr4h6ypl33mx4p7qjk2ppznr-quodlibet-full-4.6.0-unstable-2024-08-08/lib /nix/store/z187sx56kr4h6yp>
shrinking RPATHs of ELF executables and libraries in /nix/store/smvph5c6pgr9wg6k82anwyad69vrbiny-quodlibet-full-4.6.0-unstable-2024-08-08-dist
checking for references to /build/ in /nix/store/smvph5c6pgr9wg6k82anwyad69vrbiny-quodlibet-full-4.6.0-unstable-2024-08-08-dist...
patching script interpreter paths in /nix/store/smvph5c6pgr9wg6k82anwyad69vrbiny-quodlibet-full-4.6.0-unstable-2024-08-08-dist
Rewriting #!/nix/store/sd81bvmch7njdpwx3lkjslixcbj5mivz-python3-3.13.4/bin/python3.13 to #!/nix/store/sd81bvmch7njdpwx3lkjslixcbj5mivz-python3-3.13.4
wrapping `/nix/store/z187sx56kr4h6ypl33mx4p7qjk2ppznr-quodlibet-full-4.6.0-unstable-2024-08-08/bin/.exfalso-wrapped'...
Rewriting #!/nix/store/sd81bvmch7njdpwx3lkjslixcbj5mivz-python3-3.13.4/bin/python3.13 to #!/nix/store/sd81bvmch7njdpwx3lkjslixcbj5mivz-python3-3.13.4
wrapping `/nix/store/z187sx56kr4h6ypl33mx4p7qjk2ppznr-quodlibet-full-4.6.0-unstable-2024-08-08/bin/.quodlibet-wrapped'...
Rewriting #!/nix/store/sd81bvmch7njdpwx3lkjslixcbj5mivz-python3-3.13.4/bin/python3.13 to #!/nix/store/sd81bvmch7njdpwx3lkjslixcbj5mivz-python3-3.13.4
wrapping `/nix/store/z187sx56kr4h6ypl33mx4p7qjk2ppznr-quodlibet-full-4.6.0-unstable-2024-08-08/bin/.operon-wrapped'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
@nix { "action": "setPhase", "phase": "installCheckPhase" }
dbus-daemon[2208]: [session uid=1000 pid=2208] Activating service name='org.a11y.Bus' requested by ':1.0' (uid=1000 pid=2209 comm="/nix/store/sd81bvmch7njdpwx3lkjslix>
dbus-daemon[2208]: [session uid=1000 pid=2208] Successfully activated service 'org.a11y.Bus'
============================= test session starts ==============================
platform linux -- Python 3.13.4, pytest-8.3.5, pluggy-1.6.0
rootdir: /build/source
configfile: setup.cfg
plugins: xdist-3.6.1
collecting ... ^Mcollecting 0 items                                                             ^Mcollecting 54 items / 1 error                                       >

==================================== ERRORS ====================================
_________________ ERROR collecting tests/plugin/test_prefs.py __________________
quodlibet/ext/_shared/squeezebox/server.py:11: in <module>
    from telnetlib import Telnet
E   ModuleNotFoundError: No module named 'telnetlib'

The above exception was the direct cause of the following exception:
tests/plugin/test_prefs.py:13: in <module>
    from quodlibet.ext._shared.squeezebox.server import SqueezeboxError
quodlibet/ext/_shared/squeezebox/server.py:13: in <module>
    raise MissingModulePluginError("telnetlib", "Not supported in Python 3.13+") from e
E   quodlibet.plugins.MissingModulePluginError: Couldn't find module 'telnetlib'. Perhaps you need to install the package? Not supported in Python 3.13+
=============================== warnings summary ===============================
tests/plugin/test_covers.py:69
  /build/source/tests/plugin/test_covers.py:69: PytestUnknownMarkWarning: Unknown pytest.mark.flaky - is this a typo?  You can register custom marks to avoid this war>
    @pytest.mark.flaky(max_runs=3, min_passes=1, rerun_filter=delay_rerun)

tests/test_library_file.py:188
  /build/source/tests/test_library_file.py:188: PytestUnknownMarkWarning: Unknown pytest.mark.flaky - is this a typo?  You can register custom marks to avoid this war>
    @pytest.mark.flaky(max_runs=3, min_passes=2)

tests/test_util_copool.py:15
  /build/source/tests/test_util_copool.py:15: PytestUnknownMarkWarning: Unknown pytest.mark.flaky - is this a typo?  You can register custom marks to avoid this warni>
    @pytest.mark.flaky(max_runs=3, min_passes=2)

tests/test_util_thread.py:22
  /build/source/tests/test_util_thread.py:22: PytestUnknownMarkWarning: Unknown pytest.mark.flaky - is this a typo?  You can register custom marks to avoid this warni>
    @pytest.mark.flaky(max_runs=3, min_passes=2)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

ERROR: failed tests/plugin/test_prefs.py:
        D: [0000.000] util.i18n.init: LANGUAGE: None
        D: [0000.000] util.i18n.init: LANG: 'C.UTF-8'
        D: [0000.000] util.i18n.init: LANGUAGE: 'C'
        D: [0000.000] _init._init_gettext: Using local locale dir /build/source/build/share/locale
        D: [0000.000] util.i18n.register_translation: No translation found for domain 'quodlibet' in '/build/source/build/share/locale'
        D: [0000.000] util.importhelper.get_importables: Ignoring '/build/source/quodlibet/formats/__pycache__'
        D: [0000.000] formats._misc.init: MimeType=application/x-flac;audio/aac;audio/aiff;audio/dsf;audio/midi;audio/mp3;audio/mp4;audio/mpeg;audio/mpeg4;audio/mpg;a>
        D: [0000.000] formats._misc.init: Supported formats: aac, aiff, dsf, midi, monkeysaudio, mp3, mp4, mpc, spc, trueaudio, vgm, wav, wavpack, wma, xiph
        D: [0000.000] ModuleScanner.rescan: Rescanning..
        D: [0000.000] ModuleScanner.rescan: Rescanning done: 98 added, 0 removed, 4 error(s)
        D: [0000.000] util.importhelper.get_importables: Ignoring '/build/source/quodlibet/browsers/__pycache__'
        D: [0000.000] util.importhelper.get_importables: Ignoring '/build/source/quodlibet/browsers/albums/__pycache__'
        D: [0000.000] util.importhelper.get_importables: Ignoring '/build/source/quodlibet/browsers/collection/__pycache__'
        D: [0000.000] util.importhelper.get_importables: Ignoring '/build/source/quodlibet/browsers/playlists/__pycache__'
        D: [0000.000] Playlist.__init__: Playlist 'foo' initialised without library
=========================== short test summary info ============================
ERROR tests/plugin/test_prefs.py - quodlibet.plugins.MissingModulePluginError: Couldn't find module 'telnetlib...
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
================= 5 deselected, 4 warnings, 1 error in 17.66s ==================
Failed to launch bus: Failed to spawn child process ?/run/current-system/sw/bin/dbus-daemon? (No such file or directory)
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.33, NixOS, 25.11 (Xantusia), 25.11.20250617.9e83b64`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.3`
 - nixpkgs: `/nix/store/ny8c07vsrfwcb1c4i3jcpbi3qi4w9wy6-source`

### Notify maintainers

@coroa @pbogdan

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/audio/quodlibet/default.nix
+++ b/pkgs/applications/audio/quodlibet/default.nix
@@ -118,7 +118,8 @@ python3.pkgs.buildPythonApplication {
     ++ lib.optionals withMusicBrainzNgs [ musicbrainzngs ]
     ++ lib.optionals withPahoMqtt [ paho-mqtt ]
     ++ lib.optionals withPypresence [ pypresence ]
-    ++ lib.optionals withSoco [ soco ];
+    ++ lib.optionals withSoco [ soco ]
+    ++ lib.optionals (pythonAtLeast "3.13") [ standard-telnetlib ];
 
   nativeCheckInputs =
     [

--- a/pkgs/applications/audio/quodlibet/default.nix
+++ b/pkgs/applications/audio/quodlibet/default.nix
@@ -171,7 +171,7 @@ python3.pkgs.buildPythonApplication {
     gappsWrapperArgs+=(--prefix PATH : ${kakasi}/bin)
   '';
 
-  meta = with lib; {
+  meta = {
     description = "GTK-based audio player written in Python, using the Mutagen tagging library";
     longDescription = ''
       Quod Libet is a GTK-based audio player written in Python, using
@@ -187,8 +187,8 @@ python3.pkgs.buildPythonApplication {
       & internet radio, and all major audio formats.
     '';
     homepage = "https://quodlibet.readthedocs.io/en/latest";
-    license = licenses.gpl2Plus;
-    maintainers = with maintainers; [
+    license = lib.licenses.gpl2Plus;
+    maintainers = with lib.maintainers; [
       coroa
       pbogdan
     ];

